CMakeLists.txt

```diff
@@ -30,13 +28,15 @@ option(CHFL_BUILD_DOCTESTS "Build documentation tests as well as unit tests." ON
 mark_as_advanced(CHFL_BUILD_DOCTESTS)
 
 # Set a default build type if none was specified
-if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
-    message(STATUS "Setting build type to 'release' as none was specified.")
-    set(CMAKE_BUILD_TYPE "release"
-        CACHE STRING
-        "Choose the type of build, options are: none(CMAKE_CXX_FLAGS or CMAKE_C_FLAGS used) debug release relwithdebinfo minsizerel."
-    FORCE)
-    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS release debug relwithdebuginfo minsizerel none)
+if (${CMAKE_CURRENT_SOURCE_DIR} STREQUAL ${CMAKE_SOURCE_DIR})
+    if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
```

Comment:
I believe newer versions of CMake prefer ${CMAKE_BUILD_TYPE} over CMAKE_BUILD_TYPE. Maybe it's a good idea to consider migrating over for new commits?

Reply:
Yeah, seems like a good idea. As long as we can build with cmake 2.8.12 (the platform one on CentOS 7 / RedHat) I'm fine with using more modern CMake features.